### ISSUE TITLE
Works only without manually initialization

### ISSUE THREAD

**Issue by ceadreak** [jasny/bootstrap#364]
Hi, 

I have a little issue with offcanvas initialization.
Without "manually scripts", such as `$("#myMenu").offcanvas('show') ` it works great. 

But I want to initialize offcanvas with some options, such as `autohide=false`. 
`button.navbar-toggle(type="button", data-autohide="false", data-disable-scrolling="false", data-toggle="offcanvas", data-recalc="false", data-target="#sidebar", data-canvas=".canvas")`
doesn't do the job with the autohide ...

So, my question is : How to initialize offcanvas() to make it works ?

this : 
`$(document).ready(function() {
    $("#sidebar").offcanvas('show');
});`

doesn't work. The result is "nothing happens"... in Chrome debugger, I can see that DOM changes, but nothing to the screen.

Thanks,
C.

**Comment by jasny** [comment]
You need to initialize using JavaScript instead of using data attributes.

`$(function() { $("#sidebar").offcanvas({'autohide': false, 'canvas': ".canvas"}).offcanvas('show'); })`

**Comment by ceadreak** [comment]
Hi Arnold,

Thank you for your answer.
I understand the problem was I tried to initialize offcanvas() 2 times, with data attributes and js code. 

So, I removed the data attributes from the button declaration.

button.navbar-toggle(type="button", id="sidebar-toggle-button")

And I changed the js action such as that : 

var sidebar = $("#sidebar").offcanvas({
        'autohide': false,
        'canvas': ".canvas",
        'target': "#sidebar",
        'disableScrolling': false
    });

$("#sidebar-toggle-button").on('click', function() {
        sidebar.offcanvas('toggle');
    });

But it doesn’t do the trick.

The sidebar displayed great but on sidebar button click, everything disappears from the screen … 
And nothing in chrome terminal ...

So what’s wrong in the call ?? I followed the offcanvas doc and I’m using reveal method.



Thank you so much for your help.
C.
